thin-provisioning-tools: new, 0.9.0

RedL0tus opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

Introduce thin-provisioning-tools, which is required for thin-provisioning-related functionalities of LVM2.

<!-- Please input topic description here. -->

Package(s) Affected
-------------------

thin-provisioning-tools: new, 0.9.0

<!-- Please list all package(s) affected by this topic here. -->

Security Update?
----------------

<!-- If this topic is part of a security update, please uncomment "Yes,"
     and mark with the `security` label, as well as reference issue number below for priority processing. -->

<!-- Yes - Issue Number: ISSUENUMBER -->
No

<!-- Please uncomment the "Build Order" section if applicable, this is commonly needed in package updates/introduction that affects more than one package. -->

<!--
Build Order
-----------

Please describe in what order this pull request should be built.
-->

Architectural Progress
----------------------

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

Secondary Architectural Progress
--------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

Post-Merge Secondary Architectural Progress
-------------------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- TODO: CI to auto-fill architectural progress. -->

RedL0tus committed:
thin-provisioning-tools: new, 0.9.0

Signed-off-by: Kay Lin <i@v2bv.net>

RedL0tus committed:
thin-provisioning-tools: Update patches

 -  Add a new patch for updating config.guess and config.sub.

 -  Rename 0001-txt2man-Update-to-1.7.1.patch to
    0001-txt2man-update-to-1.7.1.patch

Signed-off-by: Kay Lin <i@v2bv.net>

RedL0tus committed:
thin-provisioning-tools: Update build script

Signed-off-by: Kay Lin <i@v2bv.net>

RedL0tus committed:
thin-provisioning-tools: Update patch set

Signed-off-by: Kay Lin <i@v2bv.net>

RedL0tus commented:
@cthbleachbit Kould you pwease review and test dis senpai? UwU

RedL0tus committed:
thin-provisioning-tools: Use AUTOTOOLS_DEF and AUTOTOOLS_AFTER

Signed-off-by: Kay Lin <i@v2bv.net>

RedL0tus committed:
lvm2: Add thin-provisioning-tools as PKGRECOM

Signed-off-by: Kay Lin <i@v2bv.net>

MingcongBai commented on extra-admin/thin-provisioning-tools/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+autoconf
```
Use more `abinfo` throughout`, and add verbosity.

MingcongBai commented on extra-admin/thin-provisioning-tools/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+autoconf
+
+./configure ${AUTOTOOLS_DEF} ${AUTOTOOLS_AFTER}
```
Use absolute path here.

MingcongBai commented on extra-admin/thin-provisioning-tools/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+
+make
+make DESTDIR="${PKGDIR}" BINDIR="${PKGDIR}"/usr/bin MANDIR=/usr/share/man install
+
```
So... Why did you not use an `ABTYPE`?

RedL0tus committed:
thin-provisioning-tools: Update patch set and use ABTYPE=autotools

Added several workarounds for the wicked autotools configuration
provided by the upstream.

Signed-off-by: Kay Lin <i@v2bv.net>

RedL0tus committed:
thin-provisioning-tools: Update patch set

It's original Makefile is truly cursed...

Signed-off-by: Kay Lin <i@v2bv.net>

RedL0tus committed:
thin-provisioning-tools: Update patch set and beyond script

Fix installation paths

Signed-off-by: Kay Lin <i@v2bv.net>

RedL0tus committed:
lvm2: Change thin-provisioning-tools to PKGDEP

Signed-off-by: Kay Lin <i@v2bv.net>
